dipy/reconst/shm.py

```diff
@@ -221,173 +235,288 @@ def real_sph_harm(m, n, theta, phi):
     --------
     scipy.special.sph_harm
     """
-    # dipy uses a convention for theta and phi that is reversed with respect to
-    # function signature of scipy.special.sph_harm
+    return real_sh_descoteaux_from_index(m, n, theta, phi, legacy=True)
+
+
+def real_sh_tournier_from_index(m, n, theta, phi, legacy=True):
+    """ Compute real spherical harmonics as initially defined in Tournier
+    2007 [1]_ then updated in MRtrix3 [2]_, where the real harmonic $Y^m_n$
+    is defined to be:
+
+        Real($Y^m_n$) * sqrt(2)      if m > 0
+        $Y^0_n$                      if m = 0
+        Imag($Y^|m|_n$) * sqrt(2)    if m < 0
+
+    This may take scalar or array arguments. The inputs will be broadcasted
+    against each other.
+
+    Parameters
+    ----------
+    m : int ``|m| <= n``
+        The degree of the harmonic.
+    n : int ``>= 0``
+        The order of the harmonic.
+    theta : float [0, pi]
+        The polar (colatitudinal) coordinate.
+    phi : float [0, 2*pi]
+        The azimuthal (longitudinal) coordinate.
+    legacy: bool, optional
+        If true, uses MRtrix 0.2 SH basis definition, where the ``sqrt(2)``
+        factor is omitted. Else, uses the MRtrix 3 definition presented above.
+
+    Returns
+    -------
+    real_sh : real float
+        The real harmonics $Y^m_n$ sampled at ``theta`` and ``phi``.
+
+    References
+    ----------
+    .. [1] Tournier J.D., Calamante F. and Connelly A. Robust determination
+           of the fibre orientation distribution in diffusion MRI:
+           Non-negativity constrained super-resolved spherical deconvolution.
+           NeuroImage. 2007;35(4):1459-1472.
+    .. [2] Tournier J-D, Smith R, Raffelt D, Tabbara R, Dhollander T,
+           Pietsch M, et al. MRtrix3: A fast, flexible and open software
+           framework for medical image processing and visualisation.
+           NeuroImage. 2019 Nov 15;202:116-137.
+    """
+    # In the case where m < 0, Tournier basis consider |m|
```

Comment:
# In the m < 0 case, Tournier basis considers |m|

Reply:
Done in 641c484. Thanks @CHrlS98 :+1:.

Reply:
Fixed in 641c484 :+1: